plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -422,24 +417,9 @@ private Optional<SystemTable> getViewDefinitionSystemTable(ConnectorSession sess
     @Override
     public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
-        BigQueryClient client = bigQueryClientFactory.create(session);
-        log.debug("getColumnHandles(session=%s, tableHandle=%s)", session, tableHandle);
-
         BigQueryTableHandle table = (BigQueryTableHandle) tableHandle;
-        if (table.projectedColumns().isPresent()) {
-            return table.projectedColumns().get().stream()
-                    .collect(toImmutableMap(columnHandle -> columnHandle.getColumnMetadata().getName(), identity()));
-        }
-
-        checkArgument(table.isNamedRelation(), "Cannot get columns for %s", tableHandle);
-
-        ImmutableList.Builder<BigQueryColumnHandle> columns = ImmutableList.builder();
-        columns.addAll(client.getColumns(table));
-        if (table.asPlainTable().getPartitionType().isPresent() && table.asPlainTable().getPartitionType().get() == INGESTION) {
-            columns.add(PARTITION_DATE.getColumnHandle());
-            columns.add(PARTITION_TIME.getColumnHandle());
-        }
-        return columns.build().stream()
+        checkArgument(table.projectedColumns().isPresent(), "Project columns must be present: %s", table);
```

Comment:
Can we remove Optional from `projectedColumns` if it's always present?

Reply:
No, we can't. 
https://github.com/trinodb/trino/blob/ec2a985d894e45724d16deb76efaa023eeb5c1cf/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/ptf/Query.java#L121

Reply:
So if it's allowed, shouldn't we handle case when it's empty here?